third_party/amd/lib/TritonAMDGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -1103,28 +917,41 @@ struct FpToFpOpConversion
                                    Location loc) const {
     auto srcElementType = getElementType(op.getSrc());
     auto dstElementType = getElementType(op.getResult());
+    auto roundingMode = op.getRounding();
+
+    if (srcElementType.isF32() && dstElementType.isF16()) {
+      assert(roundingMode.has_value() &&
+             "rounding mode must be specified for fp32->fp16 conversion");
+      SmallVector<Value> outVals;
```

Comment:
`outVals.reserve(operands[0].size())` to avoid potential  re-allocation frequently.

Reply:
This is `SmallVector`. So it would be optimized for holding a few elements and probably would not have the issue of frequent re-allocation?

Reply:
Depends on how many elements we have in the vector. `.reserve()` will be no-op if the total size is pretty small. In general it is a good practice and won't hurt to have them.

Reply:
Done.